plasma/root_chain/contracts/RootChain/RootChain.sol

```diff
@@ -59,34 +60,57 @@ contract RootChain {
 
     modifier incrementOldBlocks() {
         while (childChain[weekOldBlock].created_at < block.timestamp.sub(1 weeks)) {
-            if (childChain[weekOldBlock].created_at == 0) 
-                break;
-            weekOldBlock = weekOldBlock.add(1);
+            if (childChain[weekOldBlock].created_at == 0) {
+                if (childChain[nextWeekOldChildBlock(weekOldBlock)].created_at == 0)
+                    break;
+                else {
+                    weekOldBlock = nextWeekOldChildBlock(weekOldBlock);
+                }
+            }
+            else weekOldBlock = weekOldBlock.add(1);
         }
         _;
     }
 
+    function nextWeekOldChildBlock(uint256 value)
+        public
+        view
+        returns (uint256)
+    {
+        return value.div(1000).add(1).mul(1000);
```

Comment:
Why not simply `value.add(1000)`? Although this (as it stands) has the effect of setting the last 3 digits to 0.

Reply:
Unfortunately, this would not work. With such mechanism vanishing deposit would cause next child chain block number to change (by -1) - and this is what we are trying to prevent.